extensions-core/datasketches/src/test/java/org/apache/druid/query/aggregation/datasketches/theta/SketchAggregationTest.java

```diff
@@ -78,6 +85,8 @@ public class SketchAggregationTest
   @Rule
   public final TemporaryFolder tempFolder = new TemporaryFolder();
 
+  private final Closer closer;
```

Comment:
Hrm, I'm seeing the same closer pattern here too and it looks like it's building the same segments.  Q for you:

What do you think about, e.g. building the segment actually as a thing on `NestedDataUtilsTest` and using a reference-counting pattern to determine when it gets cleaned up.  That is, we statically have the test class do something like `NestedDataUtilsTest.getArraysSegmentReference()` and store the return object from that.  There's then an `@AfterClass` which counts down the reference from `NestedDataUtilsTest`.  In a "normal" static initialization, this would cause all of the tests that use the same segment to have references "checked out" from normal static initialization and then once the last one completes its `@AfterClass` that's when the cleanup would finally occur.

Reply:
same response about doing things lazy as possible on my part at the cost of overall test runtime